cli/admin/src/main/java/io/pravega/cli/admin/dataRecovery/RecoverFromStorageCommand.java

```diff
@@ -590,27 +601,30 @@ public boolean validate() throws Exception {
          *
          * @param segment Segment whose chunks need to be checked for presence.
          * @return True if all chunks of a segment are found in metadata, false otherwise.
-         * @throws Exception
          */
-        private boolean validateSegment(String segment) throws Exception {
+        protected boolean validateSegment(String segment) {
             ContainerTableExtension extension = this.container.getExtension(ContainerTableExtension.class);
             boolean isValid = true;
             try {
+                if (deletedSegments.contains(segment)) {
+                    return true;
+                }
                 List<TableEntry> entries = extension.get(NameUtils.getStorageMetadataSegmentName(this.container.getId()), Collections.singletonList(BufferView.wrap(segment.getBytes(StandardCharsets.UTF_8))), TIMEOUT).get();
                 TableEntry entry = entries.get(0);
+                if (entry == null) {
+                    return true;
+                }
                 StorageMetadata storageMetadata = SLTS_SERIALIZER.deserialize(entry.getValue().getCopy()).getValue();
                 if (storageMetadata instanceof ChunkMetadata) {
                     ChunkMetadata chunkMetdata = (ChunkMetadata) storageMetadata;
                     if (chunkMetdata.getNextChunk() != null && !chunkMetdata.getNextChunk().equalsIgnoreCase("null")) {
                         return validateSegment(chunkMetdata.getNextChunk());
-                    } else {
-                        return true;
                     }
                 }
                 if (storageMetadata instanceof SegmentMetadata) {
                     SegmentMetadata segmentMetadata = (SegmentMetadata) storageMetadata;
                     if (segmentMetadata.isActive() && segmentMetadata.getFirstChunk() != null && !segmentMetadata.getFirstChunk().equalsIgnoreCase("null")) {
-                        return validateSegment(segmentMetadata.getFirstChunk());
+                        isValid = validateSegment(segmentMetadata.getFirstChunk());
```

Comment:
Sorry if I missed, but are we using this `isValid` somewhere?

Reply:
isValid is the one that needs to be returned from this method.

Reply:
yeah, I see it now. Had missed.